opal/mca/btl/tcp/btl_tcp_component.c

```diff
@@ -467,6 +481,48 @@ static int mca_btl_tcp_component_close(void)
     return OPAL_SUCCESS;
 }
 
+/*
+ * Set an appropriate bandwidth for the btl instance. On Linux, we obtain this
+ * via an ioctl(). All other OSs, we return MCA_BTL_TCP_BTL_BANDWIDTH.
+ */
+static unsigned int
+mca_btl_tcp_ethtool_get_speed (const char *if_name)
+{
+#if !HAVE_DECL_SIOCETHTOOL
```

Comment:
You might want to change this to:

```c
#if !defined(HAVE_DECL_SIOCETHTOOL) || !defined(HAVE_DECL_STRUCT_IFREQ) || !defined(HAVE_DECL_STRUCT_ETHTOOL_CMD)
```

This is a bit pedantic, I admit, but it will cover us if we end up on a system that defines `SIOCETHTOOL`, but *doesn't* define, for example, `struct ethtool_cmd` (e.g., they pre-emptively defined `SIOCETHTOOL` and just always return `ENOTIMPL` or somesuch).  This case may never happen, but we've seen weird things like this before on esoteric systems.